site/en/docs/extensions/reference/declarativeNetRequest/index.md

```diff
@@ -51,57 +39,64 @@ a list containing dictionaries of type [Ruleset][3], as shown below.
 }
 ```
 
-### Rule Resources {: #manifest-rule-resources }
+## Concepts and Usage {: #concepts-and-usage }
 
-Rules are specified in JSON files referenced under the `"rule_resources"` manifest key.
+To use this API specify one or more rulesets. A ruleset contains an array of rules. A single rule does one of the following:
 
-Each file should contain an array of [rules](#rules) as in the [example](#example).
+-  Block a network request.
+-  Upgrade a request.
```

Comment:
Can you maybe add a short explanation what `upgrade` means (at least I don't know what upgrade means in this context).

Reply:
Done.